contracts/DCNT721A.sol

```diff
@@ -21,26 +21,33 @@ import "@openzeppelin/contracts/proxy/utils/Initializable.sol";
 import "@openzeppelin/contracts/access/Ownable.sol";
 import "./storage/EditionConfig.sol";
 import "./storage/MetadataConfig.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "./storage/DCNT721AStorage.sol";
 import "./utils/Splits.sol";
+import './interfaces/ITokenWithBalance.sol';
 
 /// @title template NFT contract
-contract DCNT721A is ERC721A, Initializable, Ownable, Splits {
-  /// ============ Immutable storage ============
-
-  /// ============ Mutable storage ============
+contract DCNT721A is ERC721A, DCNT721AStorage, Initializable, Ownable, Splits {
 
+  bool public adjustableCap;
+  bool public isSoulbound;
   uint256 public MAX_TOKENS;
   uint256 public tokenPrice;
   uint256 public maxTokenPurchase;
 
-  bool public saleIsActive = false;
+  uint256 public saleStart;
```

Comment:
I've seen Zora use uint64 for times. Not sure if this is overoptimization but wanted to bring it up: https://github.com/ourzora/zora-drops-contracts/blob/main/src/interfaces/IERC721Drop.sol#L129

Also, do we want a `publicSaleEnd`? Or, are we assuming creators will want open-ended minting times?